src/repositories/githubProjectV2.ts

```diff
@@ -0,0 +1,338 @@
+            ['grant_type', 'urn:ietf:params:oauth:grant-type:device_code'],
+          ]),
+        {
+          method: 'POST',
+          headers: [['Accept', 'application/json']],
+        }
+      )
+
+      if (!accessTokenResponse.ok) {
+        wait(deviceCode.interval)
+      }
+
+      accessToken = await accessTokenResponse.json()
+      break
+    }
+
+    if (!accessToken) {
+      throw Error('GitHub での認可に失敗しました')
+    }
+
+    await this.writeChromeStorage('token', accessToken.access_token)
+  }
+
+  async upsert(todo: ManabaTodo): Promise<void> {
+    const rel = await this.readChromeStorage('relation')
+    const link2number: Map<string, string> = new Map(rel != null ? rel : [])
+    const projectId: string = await this.readChromeStorage('target_project_id')
+    if (!projectId) {
+      throw new Error('設定が完了していないため GitHub に同期できません')
+    }
+    const [owner, repo] = await (async () => {
+      const repoFullName: string | undefined = await this.readChromeStorage(
+        'target_repo_full_name'
+      )
+      if (!repoFullName) {
+        throw new Error('設定が完了していないため GitHub に同期できません')
+      }
+      return repoFullName.split('/')
+    })()
+
+    // issue がこの拡張機能の管理下にない場合
+    if (!link2number.has(todo.link)) {
+      // issue がこの拡張機能の管理下になく、すでに完了または期限切れならスキップする
+      if (todo.status !== 'todo') {
+        return
+      }
+
+      const postIssueResponse = await this.githubRestApi<
+        {
+          owner: string
+          repo: string
+          title: string
+          body: string
+          assignees: string[]
+        },
+        { number: number; node_id: string }
+      >('POST', `/repos/${owner}/${repo}/issues`, {
+        owner,
+        repo,
+        title: `${todo.courceName}: ${todo.title}`,
+        body: `- ${todo.link}`,
+        assignees: [owner],
+      })
+      link2number.set(todo.link, postIssueResponse.number.toString())
+
+      const sdk = await this.buildGitHubGraphQlSdk()
+      const addProjectItemResponse = await sdk.AddIssueToProjectV2({
+        issueId: postIssueResponse.node_id,
+        projectId,
+      })
+
+      if (todo.due) {
+        const getDueFieldResponse = await sdk.GetField({
+          projectId,
+          fieldName: 'Due',
+        })
+
+        const fieldId = (() => {
+          const projectUnion = getDueFieldResponse.node
+          if (!projectUnion || projectUnion.__typename !== 'ProjectV2') {
+            throw new Error(`projectUnion: ${JSON.stringify(projectUnion)}`)
+          }
+          const fieldUnion = projectUnion.field
+          if (!fieldUnion || fieldUnion.__typename !== 'ProjectV2Field') {
+            throw new Error(`fieldUnion :${JSON.stringify(fieldUnion)}`)
+          }
+
+          return fieldUnion.id
+        })()
+
+        const itemId = (() => {
+          const itemId = addProjectItemResponse.addProjectV2ItemById?.item?.id
+          if (itemId == null) {
+            throw new Error('2')
```

Comment:
```suggestion
            throw new Error('<適切なエラー文>')
```

メモ: 適切なエラーなほうがうれしいという話をした
ユーザに直接は露出しないエラーとはいえ、適切なエラー文だとバグ調査などにうれしい

Reply:
このエラーは最終的に upstream で catch されてユーザに表示されるので、ちゃんとやります

Reply:
OK b5e7a40